internal/parser/parser.go

```diff
@@ -176,7 +178,7 @@ func ParseJSONText(fileName string, path tspath.Path, sourceText string) *ast.So
 	return result
 }
 
-func (p *Parser) initializeState(fileName string, path tspath.Path, sourceText string, languageVersion core.ScriptTarget, scriptKind core.ScriptKind, jsdocParsingMode scanner.JSDocParsingMode) {
+func (p *Parser) initializeState(fileName string, path tspath.Path, sourceText string, languageVersion core.ScriptTarget, scriptKind core.ScriptKind, jsdocParsingMode scanner.JSDocParsingMode, moduleResolutionKind core.ModuleResolutionKind, packageJsonType string) {
```

Comment:
Have you considered passing some struct with data instead? It's a long argument list and there are a lot of places, where an empty string is passed to `packageJsonType` var. At least a room for the future improvements.

Reply:
The main risk is that we forget to set an option (since struct literal fields are optional), but we have a tendency to "just add one more param" when we need something that's not so good either.